### DISCUSSION TITLE
Help with speed

### DISCUSSION THREAD

**Discussion by RicardoMBorges** [rdkit/rdkit#4775] [General]
Hello,
I'm no coder, but I'm trying to learn by assuming tasks. So, I'm trying to get together a pipeline for similarity comparisons using Jupyter Notebook and rdkit. It works, but when I have a bigger dataset, it becomes hard...
This is what I am doing (I would appreciate any suggestions):
```
for r in datafile.index:
    fp1 = datafile.at[r,'mfp2']
    colname = datafile.at[r,'molecule_chembl_id']
    simlist = [] #fplist
    for mol in datafile['Molecule']:
        fp = Chem.GetMorganFingerprintAsBitVect(mol,2)
        sim =DataStructs.DiceSimilarity(fp1,fp)
        simlist.append(sim)
    datafile[colname]=simlist
```
Warning:
```
<ipython-input-29-b60252549d6e>:9: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead.  To get a de-fragmented frame, use `newframe = frame.copy()`
  datafile[colname]=simlist
```
* can someone suggest me how to get this right?

Thanks

**Comment by greglandrum** [comment]
@RicardoMBorges : in order for us to help, we need to know a bit more about what you're trying to do.
It looks like you're working with a pandas dataframe. What's in that dataframe? What do you want to do with it?

**Reply by RicardoMBorges** [reply]
Thanks @greglandrum. 
Yes, I'm working on pandas dataframe. My goal is to calculate the similarity (...DiceSimilarity) across all the compounds to create a big square matrix of similarity between every compound under analysis..
Make sense?
RIcardo

**Comment by kienerj** [comment]
Maybe these 2 links help you finding a solution:

https://www.rdkit.org/docs/source/rdkit.DataManip.Metric.rdMetricMatrixCalc.html
https://iwatobipen.wordpress.com/2019/06/21/gettanimtosim-distmat-as-a-lower-triangular-matrix-and-convert-2d-array-rdkit-chemoinformatics
